### ISSUE TITLE
Puma's header normalization fixed not properly

### ISSUE THREAD

**Issue by dongvv-2538** [puma/puma#3556]
**Describe the bug**
related to the commit https://github.com/puma/puma/commit/f196b23be24712fb8fb16051cc124798cc84f70e. I understand that this will fix header like `X-Forwarded_For` will be invalidate.

This work correctly if there are more than 1 header in format `X-Forwarded_For` , `X-Forwarded-For`.
However, if there's no valid `X-Forwarded-For` header, the `HTTP_X_FORWARDED_FOR` value will still be set to value of `X-Forwarded_For`
**Puma config:**
puma v5.6.9

You can try with this test, this will be failed because `HTTP_X_FORWARDED_FOR = 2.2.2.2`

```
  def test_underscore_header_4
    hdrs = [
      "X-FORWARDED_FOR: 2.2.2.2",  # invalid, contains underscore
      "Content-Length: 5",
    ].join "\r\n"

    response = send_http_and_read "#{GET_PREFIX}#{hdrs}\r\n\r\nHello\r\n\r\n"

    assert_includes response, "HTTP_X_FORWARDED_FOR = []"
    refute_includes response, "2.2.2.2"
  end
```
**To Reproduce**
step 1: Go to web 
step 2: using some tools like BurpSuite proxy or using javascript to add invalid header like `X-Forwarded_For` or `X_Forwarded-For`
step 3: check the value of `HTTP_X_FORWARDED_FOR`

**Expected behavior**
The invalid headers should not be set as value of `HTTP_X_FORWARDED_FOR`

**Desktop (please complete the following information):**
 - OS: [e.g. Mac, Linux]
 - Puma Version [e.g. 4.1.1]

**Comment by dentarg** [comment]
Yeah, we heard from at least one user about underscores, so defaulting to rejecting them is probably out of question. https://github.com/puma/puma/issues/3502#issuecomment-2402932875

**Comment by dentarg** [comment]
Seeing https://stackoverflow.com/questions/22856136/why-do-http-servers-forbid-underscores-in-http-header-names and reading the comments makes me think this should not be the default, even in a new major version.

Should we even consider the feature?

Maybe time is better spent on https://github.com/puma/puma/issues/3397

**Comment by dentarg** [comment]
> A better solution would be to expose the underlying headers in their raw form, and I'd be strongly in favour of normalized according to the RFC, which is lower case.

That sounds more like https://github.com/puma/puma/issues/3397

So let's close this issue.
